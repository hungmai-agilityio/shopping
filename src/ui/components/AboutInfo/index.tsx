import { FONT_SIZE, FONT_WEIGHT, mada, TAG } from '@/constants';
import CardImage from '@/ui/components/Card/Image';
import { Typography } from '@/ui/components/common';
import { clsx } from 'clsx';

interface AboutInfoProps {
  desc: string;
  isReverse?: boolean;
  aboutName: string;
  src: string;
  title: string;
  styles?: string;
}

const AboutInfo = ({
  title,
  desc,
  aboutName,
  src,
  isReverse,
  styles
}: AboutInfoProps) => {
  return (
    <div
      className={clsx(
        'lg:flex block justify-between w-full gap-10',
        {
          'flex-row-reverse': isReverse
        },
        styles
      )}
    >
      <div className="lg:max-w-card-sm lg:text-left text-center">
        <Typography
          fontWeight={FONT_WEIGHT.BOLD}
          tag={TAG.H2}
          size={FONT_SIZE.MEDIUM}
          className={`${mada.className} capitalize my-5`}
        >
          {title}
        </Typography>
        <Typography className="my-6 text-dark-md" size={FONT_SIZE.X_SMALL}>
          {desc}
        </Typography>
      </div>
      <div className="flex justify-center">
        <CardImage
          src={src}
          alt={aboutName}
          width={590}
          height={349}
          title={aboutName}
          isTitleLeft={isReverse}
          className="w-full h-96"
        />
      </div>
    </div>
  );
};

export default AboutInfo;
