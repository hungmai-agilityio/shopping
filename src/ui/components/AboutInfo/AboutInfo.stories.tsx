import type { Meta, StoryObj } from '@storybook/react';

// Components
import { AboutInfo } from '@/ui/components';

const meta: Meta<typeof AboutInfo> = {
  component: AboutInfo,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
    nextjs: {
      appDirectory: true
    }
  },
  argTypes: {
    title: {
      description: 'About section title',
      control: 'text'
    },
    desc: {
      description: 'Description content',
      control: 'text'
    },
    aboutName: {
      description: 'The name of the object or event',
      control: 'text'
    },
    src: {
      description: 'Image path',
      control: 'text'
    },
    isReverse: {
      description: 'Adjust layout orientation',
      control: 'boolean'
    },
    styles: {
      description: 'Custom css',
      control: 'text'
    }
  }
};

export default meta;

type Story = StoryObj<typeof AboutInfo>;

export const Default: Story = {
  args: {
    title: 'About Our Team',
    desc: 'Meet the team behind the success of our company...',
    aboutName: 'Our Team',
    src: '/store1.webp',
    isReverse: false,
    styles: 'bg-gray-100 p-6 rounded-md'
  }
};

export const ReverseLayout: Story = {
  args: {
    src: '/store2.webp',
    title: 'Our Mission',
    aboutName: 'Mangcoding Store',
    isReverse: true,
    desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
  }
};
