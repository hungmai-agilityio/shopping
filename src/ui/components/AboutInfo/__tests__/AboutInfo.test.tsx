import { render, screen } from '@testing-library/react';

// Components
import AboutInfo from '@/ui/components/AboutInfo';

describe('About info component', () => {
  const defaultProps = {
    desc: 'description',
    isReverse: false,
    aboutName: 'Card about',
    src: '/store-1.webp',
    title: 'Product'
  };
  test('Should render component with default props', () => {
    render(<AboutInfo {...defaultProps} />);
    const title = screen.getByText(defaultProps.title);

    expect(title).toBeInTheDocument();
  });

  test('should be render match to snapshot', () => {
    const { container } = render(<AboutInfo {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
