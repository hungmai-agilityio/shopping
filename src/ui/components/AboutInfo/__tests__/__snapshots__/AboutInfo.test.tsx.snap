// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About info component should be render match to snapshot 1`] = `
<div>
  <div
    class="lg:flex block justify-between w-full gap-10"
  >
    <div
      class="lg:max-w-card-sm lg:text-left text-center"
    >
      <h2
        class="font-bold text-black text-4xl className capitalize my-5"
      >
        Product
      </h2>
      <p
        class="font-normal text-black lg:text-lg text-sm my-6 text-dark-md"
      >
        description
      </p>
    </div>
    <div
      class="flex justify-center"
    >
      <div
        class="relative inline-flex items-center justify-center"
      >
        <img
          alt="Card about"
          class="w-full h-96"
          data-nimg="1"
          decoding="async"
          height="349"
          loading="lazy"
          src="/_next/image?url=%2Fstore-1.webp&w=1200&q=75"
          srcset="/_next/image?url=%2Fstore-1.webp&w=640&q=75 1x, /_next/image?url=%2Fstore-1.webp&w=1200&q=75 2x"
          style="color: transparent; object-fit: cover;"
          width="590"
        />
        <div
          class="absolute bg-white h-14 w-11/12 text-2xl font-bold text-purple-500 right-0 p-2 overflow-hidden truncate bottom-10"
        >
          Card about
        </div>
      </div>
    </div>
  </div>
</div>
`;
